src/workerd/jsg/resource-test.c++

```diff
@@ -648,5 +648,31 @@ KJ_TEST("bundle installed works") {
   )", "string", "THIS_IS_BUILTIN_FUNCTION");
 }
 
+// ========================================================================================
+
+struct JsLazyPropertyContext: public ContextGlobalObject {
+  JSG_RESOURCE_TYPE(JsLazyPropertyContext) {
+    JSG_CONTEXT_JS_BUNDLE(BOOTSTRAP_BUNDLE);
+
+    JSG_LAZY_JS_INSTANCE_PROPERTY(bootstrapFunction, "test:resource-test-bootstrap");
+    JSG_LAZY_JS_INSTANCE_PROPERTY(BootstrapClass, "test:resource-test-bootstrap");
+  }
+};
+JSG_DECLARE_ISOLATE_TYPE(JsLazyPropertyIsolate, JsLazyPropertyContext);
+
+KJ_TEST("lazy js global function works") {
+  Evaluator<JsLazyPropertyContext, JsLazyPropertyIsolate> e(v8System);
```

Comment:
Can you include a test here where the js overrides the lazy property before it is accessed to verify that the getter is never actually called.... e.g.

```
globalThis.bootstrapFunction = 1;
console.log(bootstrapFunction);  // 1 ... rather than initializing to the function
```

Reply:
I added a test and it turns out that right now the property is not overridable. Do we actually want the ability to override builts? (is there one now?)

Reply:
For globals we have to. We've accidentally broken stuff in the past by introducing new globals that conflicted with user code

Reply:
That said, I think only some of our existing globals are overridable currently. Any *new* global we add has to be overridable tho.